packages/tse-backend/src/modules/trading-strategy/strategies/market-making/market-making.strategy.ts

```diff
@@ -0,0 +1,264 @@
+      amountChangePerLayer: command.amountChangePerLayer,
+      amountChangeType: command.amountChangeType,
+      ceilingPrice: command.ceilingPrice,
+      floorPrice: command.floorPrice,
+      status: StrategyInstanceStatus.RUNNING,
+    });
+  }
+
+  async pause(command: MarketMakingStrategyActionCommand): Promise<void> {
+    const strategyEntity: MarketMakingStrategyData =
+      await this.marketMakingService.findLatestStrategyByUserId(command.userId);
+    if (!strategyEntity) {
+      throw new BadRequestException('MarketMaking strategy not found');
+    }
+
+    await this.marketMakingService.updateStrategyStatusById(
+      strategyEntity.id,
+      StrategyInstanceStatus.PAUSED,
+    );
+    const strategy = this.strategies.get(strategyEntity.id);
+    if (strategy) {
+      clearInterval(strategy.intervalId);
+    }
+
+    this.logger.debug('Paused market making strategy');
+  }
+
+  async stop(command: MarketMakingStrategyActionCommand): Promise<void> {
+    const strategyEntity: MarketMakingStrategyData =
+      await this.marketMakingService.findLatestStrategyByUserId(command.userId);
+    if (!strategyEntity) {
+      throw new BadRequestException('MarketMaking strategy not found');
+    }
+
+    await this.marketMakingService.updateStrategyStatusById(
+      strategyEntity.id,
+      StrategyInstanceStatus.STOPPED,
+    );
+
+    const strategy = this.strategies.get(strategyEntity.id);
+    if (strategy) {
+      clearInterval(strategy.intervalId);
+    }
+
+    await this.cancelActiveOrders();
+
+    this.logger.debug(
+      'Stopped market making strategy, not filled orders have been canceled',
+    );
+  }
+
+  async delete(command: MarketMakingStrategyActionCommand) {
+    const strategyEntity: MarketMakingStrategyData =
+      await this.marketMakingService.findLatestStrategyByUserId(command.userId);
+    if (!strategyEntity) {
+      throw new BadRequestException('MarketMaking strategy not found');
+    }
+    if (
+      strategyEntity &&
+      strategyEntity.status !== StrategyInstanceStatus.STOPPED
+    ) {
+      throw new BadRequestException(
+        'MarketMaking strategy is not stopped or was deleted',
+      );
+    }
+
+    await this.marketMakingService.updateStrategyStatusById(
+      strategyEntity.id,
+      StrategyInstanceStatus.DELETED,
+    );
+
+    const strategy = this.strategies.get(strategyEntity.id);
+    if (strategy) {
+      clearInterval(strategy.intervalId);
+      this.strategies.delete(strategyEntity.id);
+    }
+
+    this.logger.debug('Soft deleted market making strategy');
+  }
+
+  async start(strategies: MarketMakingStrategyData[]): Promise<void> {
+    this.logger.debug('Starting market making strategy');
+
+    for (const strategy of strategies) {
+      if (!this.strategies.get(strategy.id)) {
+        const intervalId = setInterval(async () => {
+          await this.evaluateMarketMaking(strategy);
+        }, strategy.checkIntervalSeconds * 1000);
```

Comment:
`1000` - magic number, should be from .env

Reply:
Fixed: 254babb4